satpy/tests/reader_tests/test_li_l2_nc.py

```diff
@@ -128,6 +129,7 @@ def _test_dataset_variable(self, var_params, sname=""):
         res = self.get_variable_dataset(dataset_info, dname, handler)
         assert res.shape == shape
         assert res.dims[0] == "y"
+        assert isinstance(res.data,da.Array)
```

Comment:
was this failing before the other changes were implemented?

Reply:
I have put in commentary my implementation and run the tests and it is failing as expected 
```python 
test_li_l2_nc.py F...............................                                                                                                                                                      [100%]

================================================================================================== FAILURES ==================================================================================================
_______________________________________________________________________________________ TestLIL2.test_dataset_loading ________________________________________________________________________________________

self = <satpy.tests.reader_tests.test_li_l2_nc.TestLIL2 object at 0x7efc7f2da310>
filetype_infos = {'li_l2_af_nc': {'file_desc': {'product_type': '2-AF', 'swath_coordinates': {'azimuth': 'x', 'elevation': 'y', 'projec..._cycle:>04d}.nc'], 'file_reader': <class 'satpy.readers.li_l2_nc.LIL2NCFileHandler'>, 'file_type': 'li_l2_le_nc'}, ...}

    def test_dataset_loading(self, filetype_infos):
        """Test loading of all datasets from all products."""
        # Iterate on all the available product types:
        for ptype, pinfo in products_dict.items():
            ftype = pinfo["ftype"]
            filename_info = {
                "start_time": "0000",
                "end_time": "1000"
            }

            handler = LIL2NCFileHandler("filename", filename_info, extract_filetype_info(filetype_infos, ftype),
                                        with_area_definition=False)
            ds_desc = handler.ds_desc

            # retrieve the schema that what used to generate the content for that product:
            settings = get_product_schema(ptype)

            # Now we check all the variables are available:
            if "variables" in settings:
                self._test_dataset_variables(settings, ds_desc, handler)

                # check the sector variables:
            if "sector_variables" in settings:
>               self._test_dataset_sector_variables(settings, ds_desc, handler)

test_li_l2_nc.py:194:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
test_li_l2_nc.py:169: in _test_dataset_sector_variables
    self._test_dataset_single_sector_variable([sname, vname], desc, settings, handler)
test_li_l2_nc.py:124: in _test_dataset_single_sector_variable
    self._test_dataset_variable(var_params, sname=sname)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <satpy.tests.reader_tests.test_li_l2_nc.TestLIL2 object at 0x7efc7f2da310>
var_params = [{'name': 'event_id_north_sector', 'sector_name': 'north', 'use_rescaling': False, 'variable_name': 'event_id'}, {'def..., 'format': 'u2', 'long_name': 'ID of LI L2 Event', ...}, 'event_id', <LIL2NCFileHandler: 'filename'>, (123,), 'data/']
sname = 'north'

    def _test_dataset_variable(self, var_params, sname=""):
        """Test the validity of a given (sector) variable."""
        dataset_info, desc, dname, handler, shape, var_path = var_params
        res = self.get_variable_dataset(dataset_info, dname, handler)
        assert res.shape == shape
        assert res.dims[0] == "y"
>       assert isinstance(res.data,da.Array)
E       AssertionError: assert False
E        +  where False = isinstance(array([43325., 31613., 47855., 64957., 29417.,  5231., 23702.,  2226.,\n        1884., 21431., 29262., 52112.,  2193., ...,\n       16686., 48404., 60604., 54068., 19620., 55670., 27682., 46129.,\n       37781., 44337.,  2549.], dtype=float32), <class 'dask.array.core.Array'>)
E        +    where array([43325., 31613., 47855., 64957., 29417.,  5231., 23702.,  2226.,\n        1884., 21431., 29262., 52112.,  2193., ...,\n       16686., 48404., 60604., 54068., 19620., 55670., 27682., 46129.,\n       37781., 44337.,  2549.], dtype=float32) = <xarray.DataArray (y: 123)> Size: 492B\narray([43325., 31613., 47855., 64957., 29417.,  5231., 23702.,  2226.,\n        ... name:           event_id_north_sector\n    variable_name:  event_id\n    sector_name:    north\n    use_rescaling:  False.data
E        +    and   <class 'dask.array.core.Array'> = da.Array

test_li_l2_nc.py:132: AssertionError
```